sdk/tests/conformance2/transform_feedback/simultaneous_access.html

```diff
@@ -0,0 +1,201 @@
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+  runFeedback(gl, prog, srcVAO1, tf);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "draw should be successful");
+
+  const expected = [2, 4, 6, 8];
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected);
+
+  // this should fail because the transform feedback's buffer #0 and the
+  // srcVAO2's buffer #0 are the same buffer
+  runFeedback(gl, prog, srcVAO2, tf);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as vertex attrib and tf simultaneously");
+
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected, "should be the same as before as nothing has executed");
+
+  gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, dstBuffer);
+  runFeedback(gl, prog, srcVAO1, tf);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as uniform buffer and tf simultaneously");  
+  gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, uniformBuffer);
+
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+  gl.useProgram(prog);
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, outLoc, dstBuffer2);
+  runFeedback(gl, prog, srcVAO2, tf);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "buffer is no longer bound for transform feedback");
+
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, outLoc, dstBuffer);
+
+  // Test PIXEL_UNPACK_BUFFER
+  var tex = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, tex);
+  gl.bindBuffer(gl.PIXEL_UNPACK_BUFFER, srcBuffer1);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "PIXEL_UNPACK_BUFFER is not bound for transform feedback");
+
+  gl.bindBuffer(gl.PIXEL_UNPACK_BUFFER, dstBuffer);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, 0);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "PIXEL_UNPACK_BUFFER is bound for transform feedback");
+
+  // Test PIXEL_PACK_BUFFER
+  gl.bindBuffer(gl.PIXEL_PACK_BUFFER, srcBuffer1);
+  gl.readPixels(0, 0, 1, 1, gl.RGBA, gl.UNSIGNED_BYTE, 0);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "PIXEL_PACK_BUFFER is not bound for transform feedback");
+  gl.bindBuffer(gl.PIXEL_PACK_BUFFER, dstBuffer);
+  gl.readPixels(0, 0, 1, 1, gl.RGBA, gl.UNSIGNED_BYTE, 0);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "PIXEL_PACK_BUFFER is bound for transform feedback");
+
+  // Test bufferData family
+  gl.bindBuffer(gl.COPY_WRITE_BUFFER, dstBuffer);
+  gl.bufferData(gl.COPY_WRITE_BUFFER, 16, gl.STATIC_DRAW);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "transform feedback is not active");
+  gl.beginTransformFeedback(gl.POINTS);
+  gl.bufferData(gl.COPY_WRITE_BUFFER, 16, gl.STATIC_DRAW);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "transform feedback is active");
+  gl.endTransformFeedback();
+}
+
+function runTexImageTest() {
+  var tex = gl.createTexture();
+  gl.bindTexture(gl.TEXTURE_2D, tex);
+
+  gl.texImage2D
```

Comment:
Incomplete test.

Reply:
Good catch, removed. I hadn't noticed because the code still runs like this :open_mouth: